### PR TITLE
#8: Fixed the output dir & keys implementation

### LINKED ISSUE #7: Specify output folder
The output directory is just `/Documents` we should add an option to specify a specific output folder inside the Documents folder.

example: /Documents/cache/customcache/2020

### LINKED ISSUE #8: _keys should be saved somewhere.
Right now the `_keys` field is always empty until you save something. 

We should save all the keys. So we always know what is saved and what is not saved.

### PR DESCRIPTION
**Pull request by vanlooverenkoen**
Fixed #7 
Fixed #8

### CODE REVIEW

**Review by ikbendewilliam — APPROVED** [review]

File: lib/src/file_storage.dart [hunk 1]
```diff
@@ -27,14 +35,17 @@ class FileStorage {
     }
   }
 
-  static Future<File> write(String filename, Uint8List content) async {
+  Future<File> write(String filename, Uint8List content) async {
     final file = await _localFile(filename);
+    if (!await file.exists()) {
```
Comment (NicolaVerbeeck) [inline lib/src/file_storage.dart:40]: Should be existssync

File: lib/src/file_storage.dart [hunk 2]
```diff
@@ -18,7 +26,7 @@ class FileStorage {
     }
   }
 
-  static Future<bool> exists(String filename) async {
+  Future<bool> exists(String filename) async {
     try {
       final file = await _localFile(filename);
       return file.exists();
```
Comment (NicolaVerbeeck) [inline lib/src/file_storage.dart:32]: Could be existsSync

**Comment by NicolaVerbeeck** [comment]
Note that this change does not seem to be backwards compatible (existing items are gone)

**Comment by vanlooverenkoen** [comment]
> Note that this change does not seem to be backwards compatible (existing items are gone)

Yes indeed. Because on pub.dev we have not yet released a stable version. For me it is not a problem

**Pull request merged**